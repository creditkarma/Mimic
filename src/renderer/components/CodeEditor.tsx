// import { FormComponentProps } from "antd/lib/form";
import { highlight, languages } from "prismjs";
import "prismjs/components/prism-json";
import "prismjs/themes/prism.css";
import React from "react";
import Editor from "react-simple-code-editor";

interface IProps {
  value: string;
  onChange: (code: string) => void;
  style?: React.CSSProperties;
}

export class CodeEditor extends React.PureComponent<IProps> {
  public static defaultProps = {
    value: "",
    onChange: () => null,
  };
  public render() {
    return <div className="code-editor-container ant-input" style={this.props.style}>
      <Editor
        value={this.props.value}
        padding={8}
        onValueChange={(code) => this.props.onChange(code)}
        highlight={(code) => this.insertLineNumbers(code)}
        style={{
          lineHeight: 1.2,
          fontSize: 16,
          overflow: "visible",
        }}
      />
    </div>;
  }

  private insertLineNumbers = (code: string = ""): string =>
    highlight(code, languages.json, "json")
      .split("\n").map((line) =>
        `<span class="code-editor-line-number">${line}</span>`,
      ).join("\n")
}

export default CodeEditor;
