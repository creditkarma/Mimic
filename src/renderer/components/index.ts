export { default as EditableCell } from "./EditableCell";
export { default as MenuSwitch } from "./MenuSwitch";
import * as logos from "./logos";
export { logos };

export const trimLines = (data: string, lines: number) => {
  let iter = 0;
  let result = "";
  for (const ch of data) {
    if (ch === "\n") {
      iter += 1;
    }
    result += ch;
    if (iter === lines) {
      return result + "... trimmed";
    }
  }
  return result;
};

export const joinElements = (elements: JSX.Element[], sep: string): JSX.Element =>
  (elements as any).reduce((a: any, v: any) => [ a, sep, v ]);

export const capitalize = (word: string) =>
  word[0].toUpperCase() + word.slice(1);
