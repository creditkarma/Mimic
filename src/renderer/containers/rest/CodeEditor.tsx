import React from "react";
import { FormComponentProps } from 'antd/lib/form';
import Editor from 'react-simple-code-editor';

interface IProps extends FormComponentProps {
  value: string;
  onChange: (code: string) => void
}

export default class CodeEditor extends React.PureComponent<IProps> {
  render() {
    return (
      <Editor
        className='ant-input'
        value={this.props.value}
        padding={[8, 8, 8, 28]}
        onValueChange={code => this.props.onChange(code || '')}
        highlight={code  => this.insertLineNumbers(code)}
        style={{
          height: `calc(100vh - 420px)`,
          lineHeight: 1.2,
          fontSize: 12,
          fontFamily: '"Consolas", "Bitstream Vera Sans Mono", "Courier New", Courier, monospace'
        }}
      />
    );
  }

  insertLineNumbers(code: string): string | null {
    if (!code) {
      return null;
    }
    return code.split('\n').map(line => {
      return `<span class="code-editor-line-number">${line}</span>`
    }).join('\n');
  }
}
