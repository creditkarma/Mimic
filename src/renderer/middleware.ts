import { ipcRenderer as ipc } from "electron";
import { Middleware } from "redux";

/* tslint:disable */
const logger: Middleware = (store) => (next) => (action) => {
  console.group(action.type);
  console.info(action);
  console.groupEnd();
  return next(action);
};
/* tslint:enable */

const ipcMiddleware: Middleware = (store) => {
  ipc.on("redux-response", (e: Event, action: any) => {
    store.dispatch(action);
  });
  return (next) => (action) => {
    const type: string = action.type;
    if (type.startsWith("@@IPC_REQUEST")) {
      ipc.send("redux-request", action);
    }
    return next(action);
  };
};

export { ipcMiddleware, logger };
