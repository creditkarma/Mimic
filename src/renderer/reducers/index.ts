export * from "./clients";
export * from "./graphql";
export * from "./graphqlFiles";
export * from "./grpcFile";
export * from "./modal";
export * from "./notifications";
export * from "./requests";
export * from "./responses";
export * from "./services";
export * from "./thrift";
export * from "./thriftFile";
