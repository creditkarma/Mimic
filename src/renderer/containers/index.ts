import * as graphql from "./graphql";
import * as requests from "./requests";
import * as rest from "./rest";
import * as thrift from "./thrift";
export { default as Notifications } from "./Notifications";
export { default as ModalWindow } from "./ModalWindow";
export { graphql, rest, requests, thrift };
