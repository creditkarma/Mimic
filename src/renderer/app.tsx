import React from "react";
import * as ReactDOM from "react-dom";
import { Provider} from "react-redux";
import { Store } from "redux";
import { ModalWindow, Notifications } from "./containers";
import Routes from "./routes";

interface IProps {
  store: Store<any>;
}

const App: React.SFC<IProps> = ({store}) =>
  <Provider store={store}>
    <div>
      <Routes />
      <Notifications />
      <ModalWindow />
    </div>
  </Provider>;

export const renderApp = (props: IProps) =>
  ReactDOM.render(
    <App {...props} />,
    document.getElementById("app"),
  );

export default App;
