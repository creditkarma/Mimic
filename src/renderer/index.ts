import { renderApp } from "./app";
import configureStore from "./store";
import "./styles.css";

const store = configureStore();

renderApp({store});

if (module.hot) {
  module.hot.accept("./app", () => {
    renderApp({store});
  });
}

store.dispatch({ type: "@@IPC_REQUEST/GET_SERVICES" });
store.dispatch({ type: "@@IPC_REQUEST/GET_REQUESTS" });
