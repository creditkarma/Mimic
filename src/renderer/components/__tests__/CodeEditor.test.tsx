import React from "react";
import renderer from "react-test-renderer";
import CodeEditor from "../CodeEditor";

const JSON_CODE = `{
  "id": 1,
  "name": "John",
  "items": ["one", "two"]
}`;

describe("CodeEditor", () => {
  it("can parse JSON code", () => {
    const component = renderer.create(<CodeEditor value={JSON_CODE}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
