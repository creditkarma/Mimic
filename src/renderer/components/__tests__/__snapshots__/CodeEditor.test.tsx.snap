// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeEditor can parse JSON code 1`] = `
<div
  className="code-editor-container ant-input"
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "fontSize": 16,
        "lineHeight": 1.2,
        "overflow": "visible",
        "padding": 0,
        "position": "relative",
        "textAlign": "left",
      }
    }
  >
    <textarea
      autoCapitalize="off"
      autoComplete="off"
      autoCorrect="off"
      className="npm__react-simple-code-editor__textarea"
      data-gramm={false}
      onChange={[Function]}
      onKeyDown={[Function]}
      spellCheck={false}
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "WebkitTextFillColor": "transparent",
          "background": "none",
          "border": 0,
          "boxSizing": "inherit",
          "color": "inherit",
          "display": "inherit",
          "fontFamily": "inherit",
          "fontSize": "inherit",
          "fontStyle": "inherit",
          "fontVariantLigatures": "inherit",
          "fontWeight": "inherit",
          "height": "100%",
          "left": 0,
          "letterSpacing": "inherit",
          "lineHeight": "inherit",
          "margin": 0,
          "overflow": "hidden",
          "overflowWrap": "break-word",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "position": "absolute",
          "resize": "none",
          "tabSize": "inherit",
          "textIndent": "inherit",
          "textRendering": "inherit",
          "textTransform": "inherit",
          "top": 0,
          "whiteSpace": "pre-wrap",
          "width": "100%",
          "wordBreak": "keep-all",
        }
      }
      value="{
  \\"id\\": 1,
  \\"name\\": \\"John\\",
  \\"items\\": [\\"one\\", \\"two\\"]
}"
    />
    <pre
      aria-hidden="true"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<span class=\\"code-editor-line-number\\"><span class=\\"token punctuation\\">{</span></span>
<span class=\\"code-editor-line-number\\">  <span class=\\"token property\\">\\"id\\"</span><span class=\\"token operator\\">:</span> <span class=\\"token number\\">1</span><span class=\\"token punctuation\\">,</span></span>
<span class=\\"code-editor-line-number\\">  <span class=\\"token property\\">\\"name\\"</span><span class=\\"token operator\\">:</span> <span class=\\"token string\\">\\"John\\"</span><span class=\\"token punctuation\\">,</span></span>
<span class=\\"code-editor-line-number\\">  <span class=\\"token property\\">\\"items\\"</span><span class=\\"token operator\\">:</span> <span class=\\"token punctuation\\">[</span><span class=\\"token string\\">\\"one\\"</span><span class=\\"token punctuation\\">,</span> <span class=\\"token string\\">\\"two\\"</span><span class=\\"token punctuation\\">]</span></span>
<span class=\\"code-editor-line-number\\"><span class=\\"token punctuation\\">}</span></span><br />",
        }
      }
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "inherit",
          "display": "inherit",
          "fontFamily": "inherit",
          "fontSize": "inherit",
          "fontStyle": "inherit",
          "fontVariantLigatures": "inherit",
          "fontWeight": "inherit",
          "letterSpacing": "inherit",
          "lineHeight": "inherit",
          "margin": 0,
          "overflowWrap": "break-word",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "pointerEvents": "none",
          "position": "relative",
          "tabSize": "inherit",
          "textIndent": "inherit",
          "textRendering": "inherit",
          "textTransform": "inherit",
          "whiteSpace": "pre-wrap",
          "wordBreak": "keep-all",
        }
      }
    />
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
/**
 * Reset the text fill color so that placeholder is visible
 */
.npm__react-simple-code-editor__textarea:empty {
  -webkit-text-fill-color: inherit !important;
}

/**
 * Hack to apply on some CSS on IE10 and IE11
 */
@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
  /**
    * IE doesn't support '-webkit-text-fill-color'
    * So we use 'color: transparent' to make the text transparent on IE
    * Unlike other browsers, it doesn't affect caret color in IE
    */
  .npm__react-simple-code-editor__textarea {
    color: transparent !important;
  }

  .npm__react-simple-code-editor__textarea::selection {
    background-color: #accef7 !important;
    color: transparent !important;
  }
}
",
        }
      }
      type="text/css"
    />
  </div>
</div>
`;
